tests/fitting_test.py

```diff
@@ -43,18 +43,14 @@ def sim_data(x_min, x_max, y_func, num_x=200, binning="linear", **params):
 
 
 def compare_params(true_params, fit_params, rtol, fitter):
-    for p, v in fit_params.items():
-        # TODO: Remove
-        # if not np.isclose(v, true_params[p], rtol=rtol):
-        #     fitter.custom_plot()
-        #     plt.show()
-        assert np.isclose(v, true_params[p], rtol=rtol), p
+    for p, v in true_params.items():
+        assert np.isclose(v, fit_params[p], rtol=rtol), p
 
 
 def compare_counts(fitter):
     data_counts = np.sum(fitter.y_roi)
     model_counts = np.sum(
-        fitter.eval(fitter.x_roi, **fitter.result.best_values) * fitter.dx_roi
+        fitter.eval(fitter.x_roi, **fitter.best_values) * fitter.dx_roi
     )
     assert np.allclose(data_counts, model_counts, atol=1e-2)
```

Comment:
Is this correct? We're comparing both the model counts and data counts with their background contributions. If the background model is bad, the test will fail, despite the peak counts matching.

Reply:
Not sure if this is helpful or even what you are going for here, but it will always be true for Poisson maximum likelihood that, if the model you are fitting is able to scale itself so that sum(model) = sum(counts), then it will always do so.

Reply:
Yeah, this is one of the features of Poisson likelihood. As long as the range is the same the counts in the model and the data will be identical. If that is not the case then the fit is not a proper Poisson fit. I thought it would be a good thing to test to make sure that the fit indeed is Poisson in nature.

Reply:
😎